src/faction_camp.cpp

```diff
@@ -683,8 +683,14 @@ void talk_function::start_camp( npc &p )
     for( const auto &om_near : om_building_region( omt_pos, 3 ) ) {
         const oter_id &om_type = oter_id( om_near.first );
         if( is_ot_match( "faction_base", om_type, ot_match_type::contains ) ) {
-            popup( _( "You are too close to another camp!" ) );
-            return;
+            auto const &building_omt_pos = om_near.second;
+            auto const &camps = overmap_buffer.get_om_global( building_omt_pos ).om->camps;
+            if( std::any_of( camps.cbegin(), camps.cend(), [&building_omt_pos]( const auto & camp ) {
```

Comment:
imo auto makes this harder to read than it can be with proper types (they're pretty short in this case)

Reply:
```suggestion
            auto const &building_omt_pos = om_near.second;
            auto const &camps = overmap_buffer.get_om_global( building_omt_pos ).om->camps;
            if( std::any_of( camps.cbegin(), camps.cend(), [&building_omt_pos]( const basecamp & camp ) {
```